x/shared/session.go

```diff
@@ -106,3 +107,52 @@ func GetProofWindowCloseHeight(sharedParams *sharedtypes.Params, queryHeight int
 	return GetProofWindowOpenHeight(sharedParams, queryHeight) +
 		int64(sharedParams.GetProofWindowCloseOffsetBlocks())
 }
+
+// GetEarliestClaimCommitHeight returns the earliest block height at which a claim
+// for the session that includes queryHeight can be committed for a given supplier
+// and the passed sharedParams.
+func GetEarliestClaimCommitHeight(
+	sharedParams *sharedtypes.Params,
+	queryHeight int64,
+	claimWindowOpenBlockHash []byte,
+	supplierAddr string,
+) int64 {
+	claimWindowOpenHeight := GetClaimWindowOpenHeight(sharedParams, queryHeight)
+
+	// Generate a deterministic random (non-negative) int64, seeded by the claim
+	// window open block hash and the supplier address.
+	randomNumber := poktrand.SeededInt63(claimWindowOpenBlockHash, []byte(supplierAddr))
+
+	distributionWindowSizeBlocks := GetClaimWindowSizeBlocks(sharedParams)
+	randCreateClaimHeightOffset := randomNumber % int64(distributionWindowSizeBlocks)
```

Comment:
Since `distributionWindowSizeBlocks` ranges from window open to window close, we may end up with a `randCreateClaimHeightOffset` that is too close (1 block before `WindowCloseHeight`).

I believe `distributionWindowSizeBlocks` should be something like:
`distributionWindowSizeBlocks := GetClaimWindowSizeBlocks(sharedParams) - someSafetyBlocks`